var/spack/repos/builtin/packages/cbtf/package.py

```diff
@@ -48,12 +47,9 @@ class Cbtf(CMakePackage):
     # for rpc
     depends_on("libtirpc", type="link")
 
-    depends_on("boost@1.70.0:")
-
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
+    depends_on(
+        "boost@1.70.0:+date_time+filesystem+preprocessor+signals2+spirit_classic+test+thread"
```

Comment:
There is no variant +preprocessor in boost and no other package requests such variant:
```py
grep -r +preprocessor var/spack/repos/builtin/packages/
var/spack/repos/builtin/packages/cbtf/package.py:        "boost@1.70.0:+date_time+filesystem+preprocessor+signals2+spirit_classic+test+thread"
```

Reply:
Yup. Hence my [comment](https://github.com/spack/spack/pull/47131#issuecomment-2429299063).

Reply:
I've removed the "next-gen" boost variants so this can get merged. I need this PR to make testing easier for those changes.